packages/lib-utils/src/config.ts

```diff
@@ -0,0 +1,41 @@
+import type { ResourceFetch } from '@monorepo/common';
+
+export type UtilsConfig = {
+  /**
+   * Resource fetch implementation provided by the host application.
+   *
+   * Applications must validate the response before resolving the Promise.
+   *
+   * If the request cannot be completed successfully, the Promise should be rejected
+   * with an appropriate error.
+   */
+  appFetch: ResourceFetch;
+};
+
+let config: UtilsConfig | undefined;
+
+/**
+ * Set the {@link UtilsConfig} reference.
+ *
+ * This must be done before using any of the Kubernetes utilities.
+ */
+export const setUtilsConfig = (c: UtilsConfig) => {
+  if (config !== undefined) {
+    throw new Error('setUtilsConfig has already been called');
+  }
+
+  config = c;
+};
+
+/**
+ * Get the {@link UtilsConfig} reference.
+ *
+ * Throws an error if the reference isn't already set.
+ */
+export const getUtilsConfig = (): UtilsConfig => {
+  if (config === undefined) {
+    throw new Error('setUtilsConfig has not been called');
```

Comment:
The reason for throwing an error here is to avoid subsequent errors, e.g. `getUtilsConfig().appFetch` being undefined.

I think it's generally better to fail early when it comes to proper SDK utils initialization.